### ISSUE TITLE
[Engine/UX] The new Compare node cannot be plugged into soft_bypass

### ISSUE THREAD

**Issue by aaronventure** [christophhart/HISE#636]
I mentioned this somewhere on the forum a couple of weeks ago, but it's buried who knows where, so I'm posting here to keep better track of it. 

The same goes for the **logic_op** node. 

These are great tools for managing node processing in larger networks, but currently this can only be done by using a branch node, having an empty chain and then another chain where the target modular processing happens, then using the bool output to toggle between branches 0 and 1. 

But it's not exactly soft. Is there a bigger complication in the back that would prevent these nodes from plugging into soft_bypass?

**Comment by christoph-hart** [comment]
If you mean being able to bypass the effect that a node has on the value of the cable connection, then yes, bypass and softbypassing does not work because this only affects the audio signal that is passed through, not the parameter / modulation connections.

**Comment by aaronventure** [comment]
No, I mean controlling the bypass state of the soft bypass node with the output of a compare or a logic op node for the purposes of segmenting processing, e.g. Faust doesn't have ondemand implemented yet and there's no ETA (the branch is 2 years old), so the only way to make optimized scalable programs is to segment stuff into smaller Faust files, and then rely on HISE to do branching and/or bypassing. Compare node is an amazing tool for this (which is why I requested it), but it cannot actually control a soft bypass node. 

Current workaround is to wrap the soft bypass into chain and plug the parameter of that chain into soft bypass, or use a branch node with the first branch being an empty chain (passthrough) and the second branch being whatever processing is needed, both of which are unnecessarily messy, and I'm wary on creating designs that rely on chain parameters because [this](https://github.com/christophhart/HISE/issues/604) is still a thing.

![compare into soft bypass](https://github.com/user-attachments/assets/20ecd95a-4c72-4a80-a910-e34d6aa6868a)
